src/klein/resource.py

```diff
@@ -280,3 +280,39 @@ def write_response(r):
         d.addCallback(write_response).addErrback(log.err, _why="Unhandled Error writing response")
 
         return server.NOT_DONE_YET
+
+
+
+class KleinHTTPRequest(server.Request):
+
+    def getArg(self, key):
+        """
+        Get a single arg value.
+
+        @raises KeyError: If key doesn't exist
+        @raises ValueError: If there is more than 1 value
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        value = self.args[key]
+        if len(value) != 1:
+            raise ValueError('Too many values for: {0}'.format(key))
+        return value[0]
+
+    def getArgs(self, key):
+        """
+        Get the list of values for a key.
+
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        return self.args.get(key, [])
+
+
+
+class KleinSite(server.Site):
```

Comment:
This is a rather ugly hack on my part and I'd like to move away from it. At the time I did not know of alternative ways to overload `Request`. In retrospect, I should've created an `Request` adapter or simply append the function to the `request` object in `app.route()`. Any thoughts on how to go about this?